snuba/datasets/storage.py

```diff
@@ -21,6 +21,9 @@ class Storage(ABC):
     for more useful abstractions.
     """
 
+    def __init__(self, cluster: Cluster):
+        self.__cluster = cluster
+
```

Comment:
Why does the storage need to receive the cluster if the `get_cluster` method takes a storage key. This does not seem right since the cluster is not a structural, hardcoded property of the storage (the StorageSet would be). The mapping storageset-cluster should come from the config since the existence of a Cluster is only in config.